packages/dx-grid-core/src/plugins/table-column-resizing/reducers.ts

```diff
@@ -1,33 +1,36 @@
 import { slice } from '@devexpress/dx-core';
 import { ColumnWidthReducer } from '../../types';
-import { getColumnSize } from './helpers';
+import { getColumnsSizes } from './helpers';
 
-export const changeTableColumnWidth: ColumnWidthReducer = (
-  state, { columnName, shift, minColumnWidth, maxColumnWidth, columnExtensions,
-}) => {
+export const changeTableColumnWidth: ColumnWidthReducer = (state, payload) => {
   const { columnWidths } = state;
+  const { columnName, nextColumnName, nextColumnResizing } = payload;
   const nextColumnWidth = slice(columnWidths);
   const index = nextColumnWidth.findIndex(elem => elem.columnName === columnName);
-  const updatedColumn = nextColumnWidth[index];
-  const size = getColumnSize(
-    updatedColumn, { columnName, shift, minColumnWidth, maxColumnWidth, columnExtensions,
-    });
+  const nextIndex = nextColumnWidth.findIndex(elem => elem.columnName === nextColumnName);
+  const { size, nextSize } = getColumnsSizes(columnWidths, payload);
+
   nextColumnWidth.splice(index, 1, { columnName, width: size });
+  if (nextColumnResizing) {
+    nextColumnWidth.splice(nextIndex, 1, { columnName: nextColumnName, width: nextSize });
+  }
   return {
     columnWidths: nextColumnWidth,
   };
 };
 
-export const draftTableColumnWidth: ColumnWidthReducer = (
-  state, { columnName, shift, minColumnWidth, maxColumnWidth, columnExtensions,
-}) => {
+export const draftTableColumnWidth: ColumnWidthReducer = (state, payload) => {
   const { columnWidths } = state;
-  const updatedColumn = columnWidths.find(elem => elem.columnName === columnName)!;
-  const size = getColumnSize(
-    updatedColumn, { columnName, shift, minColumnWidth, maxColumnWidth, columnExtensions,
-    });
+  const { columnName, nextColumnName, nextColumnResizing } = payload;
+  const { size, nextSize } = getColumnsSizes(columnWidths, payload);
+  let draftColumnWidths = [{ columnName, width: size }];
+
+  if (nextColumnResizing) {
+    draftColumnWidths = [...draftColumnWidths, { columnName: nextColumnName, width: nextSize! }];
```

Comment:
I prefer to use the `return` word instead of modifying variables.

```js
   if (nextColumnResizing) {
    return { draftColumnWidths: [{ columnName, width: size }, { columnName: nextColumnName, width: nextSize! }] };
  }
  return { draftColumnWidths: [{ columnName, width: size }] }
```